github/GithubRetry.py

```diff
@@ -191,11 +203,16 @@ def get_backoff_time():
                     except (MaxRetryError, GithubException):
                         raise
                     except Exception as e:
-                        self.__log(
-                            logging.WARNING,
-                            "Failed to inspect response message",
-                            exc_info=e,
-                        )
+                        # we want to fall back to the actual github exception (probably a rate limit error)
+                        # but provide some context why we could not deal with it without another exception
+                        try:
+                            raise RuntimeError(
+                                "Failed to determine retry backoff"
+                            ) from e
+                        except RuntimeError as e:
+                            raise GithubException(
+                                response.status, content, response.headers
+                            ) from e
```

Comment:
```suggestion
                        raise GithubException(
                            response.status, "Failed to determine retry `backoff`. " + content, response.headers
                        ) from e
```

Reply:
The `GithubException` carries the response from Github (status, JSON content, headers), but the `"Failed to determine retry backoff"` bit is coming from our poor code, I think that should be separated. Github does not send that, hence the `RuntimeError`.